idaes/power_generation/carbon_capture/mea_solvent_system/unit_models/column.py

```diff
@@ -0,0 +1,1712 @@
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to build default attributes
+        super(PackedColumnData, self).build()
+
+    # ==========================================================================
+        """ Set argument values for vapor and liquid sides"""
+
+        # Set flow directions for the control volume blocks
+        # Gas flows from 0 to 1, Liquid flows from 1 to 0
+        if self.config.flow_type == FlowPattern.countercurrent:
+            set_direction_vapor = FlowDirection.forward
+            set_direction_liquid = FlowDirection.backward
+        else:
+            raise NotImplementedError(
+                "{} Packed Column class has implemented only counter-current "
+                "flow pattern. Please contact the "
+                "developer of the unit model you are using.".format(self.name))
+
+    # ==========================================================================
+        """ Build Control volume 1D for vapor phase and
+            populate vapor control volume"""
+
+        self.vapor_phase = ControlVolume1DBlock(default={
+            "transformation_method": self.config.transformation_method,
+            "transformation_scheme": self.config.vapor_side.transformation_scheme,
+            "finite_elements": self.config.finite_elements,
+            "collocation_points": self.config.collocation_points,
+            "dynamic": self.config.dynamic,
+            "has_holdup": self.config.has_holdup,
+            "area_definition": self.config.area_definition,
+            "property_package": self.config.vapor_side.property_package,
+            "property_package_args":
+                self.config.vapor_side.property_package_args})
+
+        self.vapor_phase.add_geometry(flow_direction=set_direction_vapor,
+                                      length_domain_set=self.config.length_domain_set)
+
+        self.vapor_phase.add_state_blocks(
+            information_flow=set_direction_vapor,
+            has_phase_equilibrium=False)
+
+        self.vapor_phase.add_material_balances(
+            balance_type=MaterialBalanceType.componentTotal,
+            has_phase_equilibrium=False,
+            has_mass_transfer=True)
+
+        self.vapor_phase.add_energy_balances(
+            balance_type=EnergyBalanceType.enthalpyTotal,
+            has_heat_transfer=True)
+
+        if not self.config.fix_column_pressure:
+            self.vapor_phase.add_momentum_balances(
+                balance_type=MomentumBalanceType.pressureTotal,
+                has_pressure_change=self.config.vapor_side.has_pressure_change)
+
+            # TO DO : remove this warning when there is support for deltaP
+            warnings.warn("""{} WARNING! WARNING!! WARNING!!!
+                  control volume class has not implemented a method
+                  for pressure drop. Constraint for deltaP must be provided if
+                  has_pressure_change is set to True""".format(self.name))
```

Comment:
@andrewlee94 Any idea why this is the case? CV will never implement a correlation for deltaP as far as I know. And expression or value is always provided by user.

Reply:
@PTA0001 What does this warning mean? As @jghouse88  pointed out, control volumes will never add a constraint for deltaP - that is the responsibility of the unit model.

Additionally:
1. This should use the IDAES logger, not the Python warnings module.
2. Don't repeat "WARNING" three times.

Reply:
@jghouse88  and @andrewlee94 , ```self.name```  is the unit model and this unit model has a control volume without constraint for deltaP.

Reply:
@PTA0001 Out concern was not about the unit name, it is about the fact that this message implies that control volumes will at some point add constraints for `detlaP` - this is not going to happen. It is always the unit modeler's (or user's) responsibility to add that constraint.